source/common/router/shadow_writer_impl.cc

```diff
@@ -38,5 +45,24 @@ void ShadowWriterImpl::shadow(const std::string& cluster, Http::RequestMessagePt
   thread_local_cluster->httpAsyncClient().send(std::move(request), *this, shadow_options);
 }
 
+Http::AsyncClient::OngoingRequest*
+ShadowWriterImpl::streamingShadow(const std::string& cluster, Http::RequestHeaderMapPtr&& headers,
+                                  const Http::AsyncClient::RequestOptions& options) {
+  const auto thread_local_cluster = cm_.getThreadLocalCluster(cluster);
```

Comment:
Not a blocker, but it seems like we can share a lot of the code between both streaming and shadow above e.g. when preparing the request, headers, etc. Only really differs how we use the async client

Reply:
yeah would it work to have the cluster and host setting in a shared function perhaps?

Reply:
Done.